tracing-tracy/src/lib.rs

```diff
@@ -228,40 +262,56 @@ where
                     self.stack_depth,
                 ),
                 id.into_u64(),
-            ));
+            )
+        };
+
+        TRACY_SPAN_STACK.with(|s| {
+            s.push(stack_frame);
         });
     }
 
     fn on_exit(&self, id: &Id, _: Context<S>) {
-        TRACY_SPAN_STACK.with(|s| {
-            if let Some((span, span_id)) = s.borrow_mut().pop_back() {
-                if id.into_u64() != span_id {
-                    self.client.color_message(
-                        "Tracing spans exited out of order! \
-                        Trace may not be accurate for this span stack.",
-                        0xFF000000,
-                        self.stack_depth,
-                    );
-                }
-                drop(span);
-            } else {
+        let stack_frame = TRACY_SPAN_STACK.with(|s| s.pop());
+
+        if let Some((span, span_id)) = stack_frame {
+            if id.into_u64() != span_id {
                 self.client.color_message(
-                    "Exiting a tracing span, but got nothing on the tracy span stack!",
+                    "Tracing spans exited out of order! \
+                        Trace may not be accurate for this span stack.",
                     0xFF000000,
                     self.stack_depth,
                 );
             }
-        });
+            drop(span);
+        } else {
+            self.client.color_message(
+                "Exiting a tracing span, but got nothing on the tracy span stack!",
+                0xFF000000,
+                self.stack_depth,
+            );
+        }
+    }
+
+    fn on_close(&self, id: Id, ctx: Context<'_, S>) {
+        let Some(span_data) = ctx.span(&id) else {
+            return;
+        };
+
+        if let Some(fields) = span_data.extensions_mut().get_mut::<FormattedFields<F>>() {
```

Comment:
At this point use `let-else` too?

Reply:
Sadly this doesn't format property. I think they limit the line length to 80 chars to get the one-liner.